packages/runtime-tags/src/translator/core/let.ts

```
@@ -184,3 +191,21 @@ export default {
+
+// Only primitives `===` can test are elidable: NaN never equals itself and
+// -0 is indistinguishable from 0, so both always serialize.
+function isElidableDefaultValue(
+  value: unknown,
+): value is string | number | boolean | null {
+  switch (typeof value) {
+    case "string":
+    case "boolean":
+      return true;
+    case "number":
+      return value === value && !Object.is(value, -0);
```

Biome (2.5.1)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)